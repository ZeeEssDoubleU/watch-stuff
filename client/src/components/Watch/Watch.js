import React, { Component } from "react";
import PropTypes from "prop-types";

import "./Watch.scss";
import Video from "../Video/Video.js";

class Watch extends Component {
	render() {
		return <Video id="vDjcWlCT8rg" />;
	}
}

Watch.propTypes = {};

export default Watch;

// video container
//// video
//// video metadata
//// video info

// comments container
//// total/sort comments
//// add comment
//// comment list
////// comment container
//////// comment
//////// comment info
//////// replies

// up next sidebar (right)
// view preview container
// video thumbnail
// video info
