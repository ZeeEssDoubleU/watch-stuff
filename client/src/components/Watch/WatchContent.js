import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import "./WatchContent.scss";
import Video from "../Video/Video";
import RelatedVideos from "../RelatedVideos/RelatedVideos";
import VideoInfo from "../Video/VideoInfo";
import VideoMetadata from "../Video/VIdeoMetadata";
import Comments from "../Comments/Comments";

import {
	selector_videoById,
	selector_relatedVideos,
} from "../../store/reducers/videos";
import { selector_channelDetails } from "../../store/reducers/channels";
import {
	selector_commentsByVideo,
	selector_commentsCount,
} from "../../store/reducers/comments";

const WatchContent = props => (
	<div className="watch-grid">
		<Video id={props.match.params.videoId} className="video-container" />
		<VideoMetadata className="video-metadata" video={props.video} />
		<VideoInfo
			className="video-info-container"
			video={props.video}
			channel={props.channel}
		/>
		<Comments
			className="comments-container"
			comments={props.comments}
			commentsCount={props.commentsCount}
		/>
		<RelatedVideos className="related-videos" videos={props.relatedVideos} />
	</div>
);

WatchContent.propTypes = {};

const mapStateToProps = (state, props) => ({
	video: selector_videoById(state, props.match.params.videoId),
	relatedVideos: selector_relatedVideos(state, props.match.params.videoId),
	channel: selector_channelDetails(state, props.match.params.videoId),
	comments: selector_commentsByVideo(state, props.match.params.videoId),
	commentsCount: selector_commentsCount(state, props.match.params.videoId),
});

export default withRouter(
	connect(
		mapStateToProps,
		null,
	)(WatchContent),
);
