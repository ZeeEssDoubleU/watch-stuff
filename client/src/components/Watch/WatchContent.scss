.watch-grid {
	display: grid;
	grid-template: auto auto auto 1fr / auto 402px; // 1fr row takes available left over whitespace
	padding: 24px;
	grid-gap: 8px 24px;
	max-width: 1654px;
	margin: auto;
	.video-container {
		grid-row: 1 / 2;
		grid-column: 1 / 2;
	}
	.related-videos {
		grid-row: 1 / 5;
		grid-column: 2 / 3;
	}
	.metadata {
		grid-row: 2 / 3;
		grid-column: 1 / 2;
	}
	.video-info {
		grid-row: 3 / 4;
		grid-column: 1 / 2;
	}
	.comments {
		grid-row: 4 / 5;
		grid-column: 1 / 2;
	}
	@media screen and (max-width: 1000px) {
		grid-template: auto 0.5fr auto auto 0.5fr / auto; // 1fr row takes available left over whitespace
		.video-container {
			grid-row: 1 / 2;
			grid-column: 1 / 2;
		}
		.related-videos {
			grid-row: 2 / 3;
			grid-column: 1 / 2;
		}
		.metadata {
			grid-row: 3 / 4;
			grid-column: 1 / 2;
		}
		.video-info {
			grid-row: 4 / 5;
			grid-column: 1 / 2;
		}
		.comments {
			grid-row: 5 / 6;
			grid-column: 1 / 2;
		}
	}
}
