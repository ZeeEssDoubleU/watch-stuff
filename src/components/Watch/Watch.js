import React, { useEffect } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import "./Watch.scss";
import Video from "../Video/Video";
import RelatedVideos from "../RelatedVideos/RelatedVideos";
import VideoInfo from "../Video/VideoInfo";
import VideoMetadata from "../Video/VideoMetadata";
import Comments from "../Comments/Comments";

import * as watchActions from "../../store/actions/watch";
import {
	selector_savedVideoIdsCache,
	selector_subscriptions,
} from "../../store/reducers/user";
import { selector_youtubeLibraryLoaded } from "../../store/reducers/session";
import { selector_videoById } from "../../store/reducers/videos";
import {
	selector_relatedVideos,
	selector_relatedVideosNPT,
} from "../../store/reducers/watch";
import { selector_channelDetails } from "../../store/reducers/channels";
import {
	selector_commentsByVideo,
	selector_commentsCount,
	selector_commentsNPT,
} from "../../store/reducers/comments";

const Watch = props => {
	const { pathname } = props.location;
	const { videoId } = props.match.params;
	useEffect(() => {
		if (props.youtubeLibraryLoaded) {
			props.fetchWatchDetails(videoId);
			props.updateWatchHistory(videoId);
			props.fetchRelatedVideos(videoId, props.relatedNPT, 5);
			props.fetchComments(videoId, props.commentsNPT, 5);
		}
	}, [props.youtubeLibraryLoaded, videoId]);

	return (
		<div className="watch-grid">
			<Video id={videoId} className="video-container" />
			<VideoMetadata
				className="video-metadata"
				pathname={pathname}
				video={props.video}
				saveVideo={props.saveVideo}
				savedVideos={props.savedVideos}
			/>
			<VideoInfo
				className="video-info-container"
				video={props.video}
				channel={props.channel}
				subscribe={props.subscribe}
				subscriptions={props.subscriptions}
			/>
			<Comments
				className="comments-container"
				comments={props.comments}
				commentsCount={props.commentsCount}
				fetchComments={props.fetchComments}
				videoId={videoId}
			/>
			<RelatedVideos
				className="related-vids"
				videos={props.relatedVideos}
				fetchRelatedVideos={props.fetchRelatedVideos}
				videoId={videoId}
			/>
		</div>
	);
};

const mapStateToProps = (state, props) => ({
	youtubeLibraryLoaded: selector_youtubeLibraryLoaded(state),
	video: selector_videoById(state, props.match.params.videoId),
	relatedVideos: selector_relatedVideos(state),
	relatedNPT: selector_relatedVideosNPT(state),
	channel: selector_channelDetails(state, props.match.params.videoId),
	comments: selector_commentsByVideo(state, props.match.params.videoId),
	commentsCount: selector_commentsCount(state, props.match.params.videoId),
	commentsNPT: selector_commentsNPT(state, props.match.params.videoId),
	savedVideos: selector_savedVideoIdsCache(state),
	subscriptions: selector_subscriptions(state),
});

const actionCreators = {
	fetchWatchDetails: watchActions.action_fetchWatchDetails.request,
	updateWatchHistory: watchActions.action_updateWatchHistory,
	fetchRelatedVideos: watchActions.action_fetchRelatedVideos.request,
	fetchComments: watchActions.action_fetchComments.request,
	saveVideo: watchActions.action_saveVideo,
	subscribe: watchActions.action_subscribe,
};

export default withRouter(
	connect(
		mapStateToProps,
		actionCreators,
	)(Watch),
);
