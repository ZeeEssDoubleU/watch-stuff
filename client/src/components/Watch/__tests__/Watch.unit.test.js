import React from "react";
import { shallow } from "enzyme";

import Watch from "../Watch.js";

test("renders Watch", () => {
	const wrapper = shallow(<Watch />);
	expect(wrapper).toMatchSnapshot();
});
