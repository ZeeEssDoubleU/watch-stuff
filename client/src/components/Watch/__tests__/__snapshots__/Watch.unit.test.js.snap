// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Watch 1`] = `
<Fragment>
  <Video
    id="vDjcWlCT8rg"
  />
  <RelatedVideo />
</Fragment>
`;
